core/src/apps/monero/layout/common.py

```diff
@@ -29,10 +26,7 @@
 
     while True:
         await ctx.call(ButtonRequest(code=ButtonRequestType.SignTx), ButtonAck)
-        if __debug__:
-            result = await loop.race(paginated, confirm_signal())
-        else:
-            result = await paginated
+        result = await paginated
```

Comment:
i think this should be ctx.wait?

(ISTM ctx.wait = await plus read from wire)

Reply:
We should read from wire in pretty much all places where we also wait for UI input, imho. Otherwise the host can get stuck.

Reply:
Good point. Makes sense :+1: